arch-design/dashboard/requirements_analysis_for_i_release.md

```diff
@@ -0,0 +1,124 @@
+
+
+### Requirement Analysis
+
+
+#### Multicloud - Bucket Management
+---
+* Create bucket on cloud backend (AWS, Azure, GCP and all supported vendors) based on custom tier policy
+* Delete bucket 
+* Set bucket lifecycle rules
+* Set bucket ACL.
+* Upload objects to bucket.
+* Download object from bucket.
+* Create a folder in bucket.
+* Delete folder in bucket.
+* Upload Object to folder.
+* Delete object in folder.
+* Download object from folder.
+* Archive / Restore object based on storage class
+* Copy / Paste object across buckets
+
+#### Multicloud - High Availablility
+---
+* 
+#### Delfin UI Requirements
+---
+* Register new storage drivers from SODA Dashboard
+    * VPLEX driver registration
+    * VNX driver registration
+    * IBM DS8000 driver
+    * EMC PowerMax
+    * EMC ExtremeIO
+    * IBM FlashSystem
+    * NetApp FAS
+* Add new methods of access to storage devices
+    * CLI method
+    * SMI-S method
+* Change the configure performance metrics API as per the changes made in Delfin.
+    * Two new APIs will be added.
+    * Current API will be changed; URI will be changed. Parameters will be changed.
+    * New API to get supported metrics for storage and resource (device, pool, volume).
+    * Currently only device level metrics are available.
+    * Pool level and volume level metrics will be provided. (Currently in the design phase.)
+    * Performance metrics may be different based on the driver. The new API will give the supported metrics for the driver and the dashboard visualization should allow customization by the user.(Need to discuss how this will be implemented)
+*  Add new resource types for only NAS storage device  to support NAS capabilities
+    * Qtree
+    * File Systems
+    * File share
+* Improve Grafana monitoring with additional dashboards and customization.
+
+### Input Requirements
+
+* TBD
+  
+##### Functional Requirements
+* User can create, update, delete a bucket on the cloud vendor backend from SODA Dashboard based on custom tier policy
+* User can create, delete folders on cloud vendor buckets
+* User can upload, download and delete objects to and from the cloud vendor buckets
+* User can create, update and delete lifecycle rules on buckets
+* User can archive and restore objects based on storage tier
+* User can copy / paste objects across cloud buckets
+* User is able to register new storage devices supported by Delfin
+* User is able to view new resource types for NAS storage devices
+* User is able to configure performance metric collection for storage devices and customize the Grafana dashboards to show the visualizations.
+
+
+##### Non Functional Requirements
+
+  
+## Architecture Analysis
+
+### Module Architecture
+
+
+The current SODA Multi-cloud architecture can be seen in Fig-1 below.
```

Comment:
1) This doesn't seem to be architecture but mainly flow
2) What is the difference between soda-bucket and cloud bucket?
3) So the diff between two is just the step 1), correct?
4) Does step 2 follows step 1?

Reply:
> 1. This doesn't seem to be architecture but mainly flow

The main architecture has not changed for the Dashboard. You are right. This is the flow that is depicted here. I can link to the architecture diagram [here](https://github.com/sodafoundation/architecture-analysis/blob/d1dfdc6663534c4eab3eba3f235f8b09c4362c4a/arch-design/dashboard/resources/angular-app-architecture.png)
> 2. What is the difference between soda-bucket and cloud bucket?

In Fig -1 cloud-bucket and soda-bucket makes sense. The existing SODA multilcoud architecture requires the user to create bucket on the cloud using the cloud console and provide the bucket name as an input during the Backend registration process in SODA. The soda-bucket refers to all the buckets created within SODA on the registered backends. These are in fact folders that are created within the cloud-buckets.
After the changes that have been proposed are implemented the step to create a cloud bucket can be performed through SODA multicloud itself. Here there would be no need for the user to create a bucket and provide the name as part of backend registration. During Iter2_I the backend registration was not changed and that is why this has remained here. 
Will be fixed.

> 3. So the diff between two is just the step 1), correct?

Correct. In the new changes there will be no difference in cloud-bucket and soda-bucket. All buckets that user creates from SODA will be actual buckets and not folders before.

> 4. Does step 2 follows step 1?

This is slightly confusing. These numbers do not signify order (although numbers usually do.)
I will fix these and update the diagrams. 

@PravinRanjan10 we should look at the diagrams for the multicloud sequence diagrams again and make the changes discussed above.

Reply:
@kumarashit This has been updated. Please check the updated diagrams. These have been added to the document as well.

Current .
![multi_cloud_architecture_current](https://user-images.githubusercontent.com/19162717/113424932-95d20280-93ee-11eb-82cb-a89002f041cd.png)

Updated.
![multi_cloud_architecture_updated](https://user-images.githubusercontent.com/19162717/113424940-979bc600-93ee-11eb-8683-9cf38d282eb8.png)